guides/v2.1/cloud/cdn/fastly-vcl-badreferer.md

```diff
@@ -11,57 +11,104 @@ functional_areas:
 
-## Create badreferer.json {#vcl}
+## Create a custom VCL snippet to block referrer spam
 
-For this example, you may only want to run it against the Production server. You can also add it to Staging for testing.
+The following custom VCL snippet code (JSON format) checks if incoming requests are from a referrer site included in the `referrer_blocklist` edge dictionary. If the site is in the block list, the request is blocked.
 
-Create an `badreferer.json` file with the following JSON content:
 
 ```json
 {
-  "name": "badreferer",
+  "name": "block_bad_referrer",
   "dynamic": "0",
   "type": "recv",
   "priority": "5",
   "content": "set req.http.Referer-Host = regsub(req.http.Referer, \"^https?://?([^:/\\s]+).*$\", \"\\1\"); if (table.lookup(referer_blocklist, req.http.Referer-Host)) { error 403 \"Forbidden\"; }"
 }
 ```
+Review the code to determine if you need to change any values:
+
+  -  `name`: Name for the VCL snippet. For this example, we used the name `block_bad_referrer`.
+  
+  -  `dynamic`: Value 0 indicates that this is a [regular snippet](https://docs.fastly.com/guides/vcl-snippets/using-regular-vcl-snippets)  that you upload to the versioned VCL for the Fastly configuration.
+
+  -  `priority`: Determines when the VCL snippet runs. The priority is set to 5, which means that this snippet code runs before any of the default Magento VCL snippets (magentomodule) that have a priority of 50.
+
+  -  `type`: This VCL is a `recv` snippet type which adds the snippet code to the `vcl_recv`subroutine below the boilerplate VCL and above any objects.
+ 
+  -  `content`: When the VCL code in this example runs, it captures the host of a referer website into a header, and then compares the host name to the list of URLs in the `referrer_blocklist` dictionary. If the host name matches, the request is blocked with a `403 Forbidden` error. See the [Fastly VCL reference](https://docs.fastly.com/vcl/reference/) for information about creating Fastly VCL code snippets.
+
+You can add the custom VCL snippet to your Fastly service configuration from the Admin UI (requires Fastly module 1.2.58 or later). If you do not have access to the Admin UI, you can save the JSON code example in a file and upload it using the Faslty API. See [Creating a VCL snippet using the Fastly API](https://docs.fastly.com/vcl/vcl-snippets/using-regular-vcl-snippets/#via-the-api).
+
+#### To add the custom VCL snippet from the Admin UI
+
+1.  From the Magento Admin UI, navigate to **Stores** > **Configuration** > **Advanced** > **System** > **Fastly Configuration**.
+
+1.  Expand the **Custom VCL snippet** section.
+
+1.  Click **Create Custom Snippet**.
+
+1.  Complete the snippet form as shown in the following figure:
+
+    ![Create custom referrer block VCL snippet]
```

Comment:
so the snippet you upload needs the same escaping required in the JSON file?   In the Fastly documentation, it looked like the code you add through the UI doesn't need escaping.  I wondered if the escaping is required  only if you are submitting the data in a cURL payload?

Reply:
Escaping is only required when you upload it as JSON using the API. You're correct that when you use the Magento admin panel you shouldn't escape the snippet. 

You have a picture of the admin panel as an example to show users. This is good and the code in that picture is correct. It doesn't have JSON escaping and that's good, it shouldn't have it. There's no problem in the picture, I just think that it would be nice if people could copy and paste that code into their own admin panel. They can't copy and paste from the image, though.

Reply:
There's no need to have everything able to be copied and pasted, so if you don't think it's necessary then I'm good with this as-is.